wrappers/csharp/unit-tests/TestManaged.cs

```diff
@@ -340,5 +339,43 @@ public void VerifyPassword()
         {
             Assert.IsTrue(Managed.VerifyPassword(TestData.BytesTestKey, TestData.TestHash));
         }
+
+        [TestMethod]
+        public void GenerateSharedKey()
+        {
+            const int nbShares = 5;
+            const int secretLength = 10;
+            const int threshold = 3;
+            var shares = new byte[nbShares][];
+            var result = Managed.GenerateSharedKey(nbShares, threshold, secretLength, ref shares);
+            Assert.AreEqual(0, result);
+        }
+
+        [TestMethod]
+        public void JoinShares()
+        {
+            var shares = GetSharesKeys();
+            var result = Managed.JoinShares(shares);
+            Assert.AreEqual(0, result);
```

Comment:
It should check that a shared key is created and that the same secret key can be generated using multiple pairs of shares.